### PR TITLE
chore: create `orm` for specific cases

### PR DESCRIPTION
**Pull request by ggazzo**
[OPI-20]
https://adr.rocket.chat/0099

## Further comments

[OPI-20]: https://rocketchat.atlassian.net/browse/OPI-20?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/models/src/updater.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		}
+
+		this.dirty = true;
+
+		try {
+			await this.model.updateOne(query, {
+				...(this._set && { $set: Object.fromEntries(this._set) }),
+				...(this._unset && { $unset: Object.fromEntries([...this._unset.values()].map((k) => [k, 1])) }),
+				...(this._inc && { $inc: Object.fromEntries(this._inc) }),
+				...(this._addToSet && { $addToSet: { $each: Object.fromEntries(this._addToSet) } }),
```
Comment (sampaiodiego) [inline packages/models/src/updater.ts:70]: this code is used multiple times, what if you move it to a private method and use it instead?

File: packages/models/src/updater.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	unset<K extends keyof UnsetProps<T>>(key: K): Updater<T> {
+		this._unset = this._unset ?? new Set<keyof UnsetProps<T>>();
+		this._unset.add(key);
+		return this;
+	}
+
+	test1<K extends UpdateFilter<T>>(key: K) {
+		console.log(key);
+	}
+
```
Comment (sampaiodiego) [inline packages/models/src/updater.ts:37]:
```suggestion
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: packages/models/src/updater.ts (hunk 1 above)
Comment (ggazzo) [reply packages/models/src/updater.ts:66]: image

**Review by sampaiodiego — APPROVED** [review]
nojo

**Pull request merged**